client/opentracing/src/main/java/io/opentracing/APMSpan.java

```diff
@@ -59,22 +59,22 @@
      * @param builder  The span builder
      * @param recorder The trace recorder
      */
-    public APMSpan(APMSpanBuilder builder, TraceRecorder recorder, ContextSampler sampler) {
+    public APMSpan(APMSpanBuilder builder, TraceRecorder recorder, ContextSampler sampler, String serviceName, String buildStamp) {
```

Comment:
We have two approaches here: one is to propagate the `serviceName` and `buildStamp` from the `APMTracer` down to the `TraceContext`, even duplicating this along the way. The other possible solution is to wrap the `TraceRecorder` into a `EnvironmentAwareTraceRecorder`, which would contain the logic for parsing the `serviceName` and `buildStamp`. IMO, it would be cleaner, but it's more complex and would probably not perform as good as the current solution (although it wouldn't necessarily be bad). 

What do you think?